plugins/groups/src/components/GroupView/components/Members.tsx

```diff
@@ -0,0 +1,130 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import Avatar from 'react-avatar';
+import Alert from '@material-ui/lab/Alert';
+import {
+  Box,
+  Card,
+  createStyles,
+  Grid,
+  Link,
+  makeStyles,
+  Typography,
+} from '@material-ui/core';
+import { InfoCard, Progress, useApi } from '@backstage/core';
+import { UserEntity } from '@backstage/catalog-model';
+import { Link as RouterLink, generatePath, useParams } from 'react-router-dom';
+import { catalogApiRef } from '@backstage/plugin-catalog';
+import { useAsync } from 'react-use';
+import { viewMemberRouteRef } from '../../../plugin';
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    infoCard: {
+      '& .sb-avatar': {
+        position: 'absolute',
+        top: '-25px',
+      },
+    },
+    card: {
+      overflow: 'visible',
+      position: 'relative',
+    },
+  }),
+);
+
+const MemberComponent = ({ member }: { member: UserEntity }) => {
+  const classes = useStyles();
+  const { name: metaName } = member.metadata;
+  const { profile } = member.spec;
+
+  return (
+    <Grid item md={3}>
+      <Card raised classes={{ root: classes.card }}>
+        <Box
+          display="flex"
+          flexDirection="column"
+          m={3}
+          alignItems="center"
+          justifyContent="center"
+        >
+          <Avatar
+            name={profile?.displayName}
+            src={profile?.picture}
+            size="50px"
+            round
+          />
+          <Box py={4} textAlign="center">
+            <Typography variant="h5">
+              <Link
+                component={RouterLink}
+                to={`/groups/${generatePath(viewMemberRouteRef.path, {
+                  memberName: metaName,
+                })}`}
+              >
+                {profile?.displayName}
+              </Link>
+            </Typography>
+            <Typography variant="caption">{profile?.email}</Typography>
+          </Box>
+        </Box>
+      </Card>
+    </Grid>
+  );
+};
+
+export const Members = () => {
+  const classes = useStyles();
+  const { groupName } = useParams();
+  const catalogApi = useApi(catalogApiRef);
+  const { loading, error, value: members } = useAsync(async () => {
+    const membersList = await catalogApi.getEntities({
+      filter: {
+        kind: 'User',
+      },
+    });
+    const groupMembersList = ((membersList.items as unknown) as Array<
+      UserEntity
+    >).filter(member => member.spec.memberOf.includes(groupName));
```

Comment:
You should avoid consuming the `memberOf`, `parent` or `children` properties in the spec. Try to use the generated [relations](https://backstage.io/docs/features/software-catalog/well-known-relations) instead. They are based on the properties in the spec. But I'm not sure if everything is ready for this yet, this is active in development right now.

Reply:
It's also going to be nice when we (some time in the near future) can do the by-relation filtering already in the getEntities above, so we don't have to resort to manual grep here

Reply:
> I think we're at the point where we should. The relations are emitted now, as far as I know.

Yeah, they are available and work well in the frontend. Just working on the providesApi and consumesApi stuff, the relations are perfect 🙂 👍

Reply:
I just try them, works great 👍